snap/squashfs/squashfs_test.go

```diff
@@ -236,6 +236,58 @@ func (s *SquashfsTestSuite) TestUnpackGlob(c *C) {
 	c.Assert(osutil.FileExists(filepath.Join(outputDir, "meta/snap.yaml")), Equals, false)
 }
 
+func (s *SquashfsTestSuite) TestUnpackDetectsFailures(c *C) {
+	mockUnsquashfs := testutil.MockCommand(c, "unsquashfs", `
```

Comment:
does it really look like this in real life?
if it outputs this much (and maybe even if not), I'd suggest changing the check: make a discarding `io.Writer` that detects the failure, and use that as `cmd.Stderr`.

Reply:
Indeed, thanks for this suggestion. It will potentially return an error for each file it tries to unpack (e.g. on out of disk space) - so potentially a lot of output. I created a custom writer to minimize the impact. There is a tradeoff here, the code is not as easy anymore, feedback welcome. We might consider keeping the simplistic approach because we generally do not unpack many files in snapd (bootloader asserts, kernel/initramfs) so we shouldn't generate insane amount of output (currently at least).